fleet_adapter_mir_actions/fleet_adapter_mir_actions/rmf_wait_until.py

```diff
@@ -0,0 +1,483 @@
+                        f'{supported_signal_types}. Unable to instantiate '
+                        f'WaitUntil MirActionFactory.'
+                    )
+
+                # Pass signal config into verify functions to validate types
+                # and catch missing values
+                match signal_type:
+                    case 'mission':
+                        self.verify_mission(
+                            signal_config.get('mission_name'),
+                            signal_config.get('retry_count'),
+                            signal_config.get('resubmit_on_abort')
+                        )
+                    case 'plc':
+                        self.verify_plc(signal_config.get('register'))
+                    case 'custom':
+                        self.verify_custom_module(signal_config.get('module'))
+        else:
+            # If the user did not provide any valid signal config, log a
+            # warning to remind users to provide signal type config in any task
+            # description they submit
+            self.context.node.get_logger().warn(
+                f'WaitUntil ActionFactory is instantiated for robot '
+                f'[{self.context.name}], but no valid signals config has been '
+                f'provided in the action config! Any move off signal config '
+                f'will have to be provided in the task description submitted '
+                f'to RMF.'
+            )
+            return
+
+        # Register default signal type
+        default_signal = context.action_config.get('default_signal')
+        if default_signal is not None:
+            if default_signal not in supported_signal_types:
+                raise ValueError(
+                    f'User provided a default signal type {default_signal} '
+                    f'in the action config for WaitUntil MirAction, but the '
+                    f'default signal type is not supported!')
+            elif default_signal not in signals.keys():
+                raise ValueError(
+                    f'User provided a default signal type {default_signal} '
+                    f'in the action config for WaitUntil MirAction, but the '
+                    f'default signal type is not configured!')
+        else:
+            self.context.node.get_logger().warn(
+                f'WaitUntil ActionFactory instantiated for robot '
+                f'[{self.context.name}], but no default signal type has been '
+                f'provided in the action config! If no signal type is '
+                f'populated in the task description, there will be no move '
+                f'off signal available and the robot will wait for the full '
+                f'duration of the timeout when this action is triggered.'
+            )
+
+    def supports_action(self, category: str) -> bool:
+        match category:
+            case 'wait_until':
+                return True
+            case _:
+                return False
```

Comment:
nit: not to overuse match-case statements where unnecessary
```suggestion
        return True if category == 'wait_until' else False
```